src/services/auth.js

```diff
@@ -48,6 +49,20 @@ passport.use(new HTTPSignatureStrategy(
       })
   }))
 
+passport.use(new ClientCertStrategy((fingerprint, info, done) => {
+  if (!config.auth.client_certificates_enabled) {
+    return done(new UnauthorizedError('Unsupported authentication method'))
+  }
+
+  Account.findByFingerprint(fingerprint)
+    .then(function (userObj) {
+      if (!userObj || userObj.is_disabled || userObj.fingerprint !== fingerprint) {
```

Comment:
It might be safer to fail authorization if the fingerprint is missing/empty in `userObj`.